pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -464,6 +482,24 @@ func (gc *GarbageCollector) isDangling(ctx context.Context, reference metav1.Own
 func (gc *GarbageCollector) classifyReferences(ctx context.Context, item *node, latestReferences []metav1.OwnerReference) (
 	solid, dangling, waitingForDependentsDeletion []metav1.OwnerReference, err error) {
 	for _, reference := range latestReferences {
+		// If the resource type of reference is not synced yet, we should return error to
+		// requeue the item.
+		gv, err := schema.ParseGroupVersion(reference.APIVersion)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		rm, err := gc.restMapper.RESTMapping(
+			schema.GroupKind{
+				Group: gv.Group,
+				Kind:  reference.Kind,
+			}, gv.Version)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		if !gc.dependencyGraphBuilder.IsResourceSynced(rm.Resource) {
+			return nil, nil, nil, ownerResourceNotSyncedError
+		}
```

Comment:
also, resolving restmapping can be slow, so we don't want to put it in this code path... isDangling checks a negative cache to avoid resolving restmapping repeatedly for the same references

Reply:
Thanks for your reply! Agree with your sugguestion, isDangling should do the same work.